sourcegraph/sourcegraph
# source: redesign "Add code host connection" page

Closes [#55010](https://github.com/sourcegraph/sourcegraph/issues/55010)

This PR implements the [redesign of the code host connection page](https://www.figma.com/file/tJ9LAWTljukeunwBZLKDbE/Cleanup-add-code-host-connection-page?type=design&node-id=0-1&mode=design&t=K4TvFAJU147IQXmX-0).
The PR introduces the concept of grouping across external services so we can prioritize displaying popular code hosts. The GIF below shows a glimpse of what the page looks like and how it works.

https://github.com/sourcegraph/sourcegraph/assets/25608335/389dde2c-6fea-4277-807a-f3a9cb1ed2d9

We also have a condition where we [filter out some code hosts based on the license on the Sourcegraph instance](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/components/externalServices/AddExternalServicesPage.tsx?L93). I cleaned up the logic a bit, so it's easier to understand what's going on.

Below is a screenshot of what it looks like now:

<img width="1218" alt="CleanShot 2023-08-07 at 20 54 24@2x" src="https://github.com/sourcegraph/sourcegraph/assets/25608335/c80f475c-46d2-4e06-b98c-30e6bdf21337">

## Test plan



* Updated the `AddExternalServicesPage` Storybook to include a case where the license is `business-0`
* Added snapshot tests for the `ExternalServiceGroup` component.

#### Manual Testing

* On your Sourcegraph instance, navigate to the Site Admin section.
* Click on `Code host connection` section on the side bar.
* Click on the `Add code host connection` button
* The new UI should be accessible.
